### PR TITLE
Save ship outfits in alphabetic order to savegame.

### PR DESCRIPTION
**Pull request by petervdmeer**
**Feature:** This PR implements part of the feature request detailed and discussed in issue #5171

## Feature Details
This PR adds a temporary map in the function that saves the ship-outfits to a save-game file to ensure that the outfits are saved in alphabetical order.

## UI Screenshots
N/A

## Usage Examples
N/A, the game automatically applies the ordering when saving.

## Testing Done
I started the game with an existing save-game, pressed depart and then closed the game. The outfits in the save-game before and after starting where the same (but the outfits after starting the game were sorted in alphabetical order).

## Performance Impact
The effect of this change is that the function that saves the game takes a little bit longer time and temporary uses a little bit more memory (since it needs to build up the temporary map). But ships have small lists of outfits, so the impact should be very minimal.
Saving the game is not performance critical code.

### CODE REVIEW

File: source/Ship.cpp [hunk 1]
```diff
@@ -704,13 +704,23 @@ void Ship::Save(DataWriter &out) const
...
+			// The game-engine doesn't care in which order outfits are
+			// saved or loaded, the sorting here is done as a service for
+			// content creators that use savegames for working on their
+			// new content.
+			map<string, int> orderedOutfits;
 			for(const auto &it : outfits)
 				if(it.first && it.second)
+					orderedOutfits[it.first->Name()] = it.second;
+			for(const auto &ito : orderedOutfits)
 				{
```
Comment (tehhowch) [inline source/Ship.cpp:719]: Braces are over-indented

File: source/Ship.cpp [hunk 2]
```diff
@@ -704,13 +704,23 @@ void Ship::Save(DataWriter &out) const
...
-					if(it.second == 1)
-						out.Write(it.first->Name());
+					if(ito.second == 1)
```
Comment (tehhowch) [inline source/Ship.cpp:720]: No need to change the local variable name from the common `it` to `ito`

File: source/Ship.cpp (hunk 1 above)
Comment (petervdmeer) [reply source/Ship.cpp]: Sorry, was an editting issue. Fixed now.

File: source/Ship.cpp (hunk 2 above)
Comment (petervdmeer) [reply source/Ship.cpp]: I thought it was more clear for somebody reading the code, but it indeed doesn't really make a difference. Changed back to it.

**Pull request merged**